client/components/ReviewsRender.jsx

```diff
@@ -1,11 +1,65 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+const wordConditionFinder = (text, filterCondition) => {
+  const results = [];
+  const findAllInstances = (word) => {
+    const lowerCaseWord = word.toLowerCase();
+    const lowerFilterCondition = filterCondition.toLowerCase();
+    if (lowerCaseWord.indexOf(lowerFilterCondition) === -1) {
+      return results;
+    }
+    results.push(
+      word.substring(lowerCaseWord.indexOf(lowerFilterCondition),
+        lowerCaseWord.indexOf(lowerFilterCondition) + filterCondition.length),
+    );
+    return findAllInstances(lowerCaseWord.slice(lowerCaseWord.indexOf(lowerFilterCondition) + 1));
```

Comment:
Calling lowerCaseWord.slice(lowerCaseWord.indexOf(lowerFilterCondition) m times means O(m*n) where n === lowerCaseWord.length - lowerCaseWord.indexOf(lowerFilterCondition) + 1.
Consider passing the computed index (lowerCaseWord.indexOf(lowerFilterCondition) + 1) and using it in lowerCaseWord.indexOf during the next recursion.